src/services/api/workspace.ts

```diff
@@ -70,6 +71,7 @@ export async function createWorkspaceFromDevfile(
   const attrs = Object.keys(attributes).map(key => `${key}:${attributes[key]}`);
 
   try {
+    // TODO rework this solution with WorkspaceClient
```

Comment:
It depends on the issue about updating Workspace Client, right? If so - please add it there https://github.com/eclipse/che/issues/17700

Reply:
I have done:
...
```
Found errors(IResources):

start a workspace method does not work corectly;
createWorkspaceFromDevfile - doesn`t exist;
getKubernetesNemespace - doesn`t exist;
getDevfileSchema - doesn`t exist;
getFactoryResolver - doesn`t exist;
```